projects/batfish/src/main/java/org/batfish/smt/PropertyChecker.java

```diff
@@ -197,10 +205,45 @@ public static AnswerElement computeReachability(IBatfish batfish, HeaderLocation
       enc.add(enc.mkEq(f, enc.mkInt(0)));
     }
 
-    VerificationResult res = enc.verify();
+    Tuple<VerificationResult, Model> result = enc.verify();
+    VerificationResult res = result.getFirst();
+    Model model = result.getSecond();
     // res.debug(enc.getMainSlice(), true, null);
-    SmtOneAnswerElement answer = new SmtOneAnswerElement();
+
+    // Analyze the result to see if we can figure out what went wrong
+    Set<String> unreachableSources = new HashSet<>();
+    Map<String, Boolean> diffReachable = new HashMap<>();
+    if (!res.isVerified()) {
+      if (q.getEquivalence()) {
+        assert (reach2 != null);
+        for (String source : sourceRouters) {
+          BoolExpr sourceVar1 = reach.get(source);
+          BoolExpr sourceVar2 = reach2.get(source);
+          String val1 = evaluate(model, sourceVar1);
+          String val2 = evaluate(model, sourceVar2);
+          boolean notEqual =
+              ((val1 == null && val2 != null) || (val1 != null && !val1.equals(val2)));
```

Comment:
there's a built in library function for this: `!Objects.equals(val1, val2)`

Reply:
That's convenient, thanks.